### PR TITLE
Revise Svc::FramingProtocol

### PR DESCRIPTION
**Pull request by bocchino**
* Clean up include guards and banner comments
* Add missing comments to interface symbols
* Replace C preprocessor symbols with C++ type and constant symbols, per coding guidelines
* Add `DEFRAMING_INVALID_FORMAT` to cover the case of an invalid format, e.g., a bad start word

### CODE REVIEW

**Comment by LeStarch** [comment]
Does this address: https://github.com/nasa/fprime/issues/1296?

**Comment by bocchino** [comment]
This PR does not address #1296.

**Comment by bocchino** [comment]
Deframer is on my list to review, and I could address #1296 at that time.

File: Svc/FramingProtocol/FprimeProtocol.hpp [hunk 1]
```diff
@@ -10,33 +10,66 @@
...
+      FprimeDeframing();
+
+      //! Validates data against the stored hash value
+      //! 1. Computes the hash value V of bytes [0,SIZE-1] in the circular buffer
+      //! 2. Compares V against bytes [SIZE, SIZE + HASH_DIGEST_LENGTH - 1] of
+      //!    the circular buffer, which are expected to be the stored hash value.
+      bool validate(Types::CircularBuffer& buffer, U32 size);
+
+      //! Implements the deframe method
+      DeframingStatus deframe(
```
Comment (arizvi786) [inline Svc/FramingProtocol/FprimeProtocol.hpp:67]: Add comments for input arguments

File: Svc/FramingProtocol/FprimeProtocol.hpp [hunk 2]
```diff
@@ -10,33 +10,66 @@
...
+  class FprimeDeframing : public DeframingProtocol {
+    public:
+
+      //! Constructor
+      FprimeDeframing();
+
+      //! Validates data against the stored hash value
+      //! 1. Computes the hash value V of bytes [0,SIZE-1] in the circular buffer
+      //! 2. Compares V against bytes [SIZE, SIZE + HASH_DIGEST_LENGTH - 1] of
+      //!    the circular buffer, which are expected to be the stored hash value.
```
Comment (arizvi786) [inline Svc/FramingProtocol/FprimeProtocol.hpp:63]: Add comments for input arguments

File: Svc/FramingProtocol/FprimeProtocol.hpp [hunk 3]
```diff
@@ -10,33 +10,66 @@
...
+  }
+
+  //! \brief Implements the F Prime framing protocol
+  class FprimeFraming: public FramingProtocol {
+    public:
+
+      //! Constructor
+      FprimeFraming();
+
+      //! Implements the frame method
```
Comment (arizvi786) [inline Svc/FramingProtocol/FprimeProtocol.hpp:44]: Add comments for input arguments

File: Svc/FramingProtocol/DeframingProtocol.hpp [hunk 4]
```diff
@@ -41,6 +41,7 @@ class DeframingProtocol {
         DEFRAMING_INVALID_SIZE, /*!< Invalid size found */
         DEFRAMING_INVALID_CHECKSUM, /*!< Invalid checksum */
         DEFRAMING_MORE_NEEDED, /*!< Successful deframing likely with more data */
+        DEFRAMING_INVALID_FORMAT, /*< Invalid format */
         DEFRAMING_MAX_STATUS
```
Comment (arizvi786) [inline Svc/FramingProtocol/DeframingProtocol.hpp:45]: Add comment for last enum value

File: Svc/Deframer/test/ut-unified/Tester.hpp [hunk 5]
```diff
@@ -140,7 +140,7 @@ class Tester : public DeframerGTestBase {
     bool m_garbage;
     Fw::ComPacket::ComPacketType m_uplink_com_type;
     // Initialize to empty list to appease valgrind
-    U8 m_uplink_data[(sizeof(FP_FRAME_TOKEN_TYPE) * 2) + sizeof(U32) + FW_COM_BUFFER_MAX_SIZE] = {};*/
+    U8 m_uplink_data[(sizeof(FpFrameHeader::TokenType) * 2) + sizeof(U32) + FW_COM_BUFFER_MAX_SIZE] = {};*/
```
Comment (arizvi786) [inline Svc/Deframer/test/ut-unified/Tester.hpp:143]: Removed unused code in comment block

**Review by arizvi786 — COMMENTED** [review]
@bocchino good job, just a few minor comments, please take a look

File: Svc/FramingProtocol/DeframingProtocolInterface.hpp [hunk 6]
```diff
@@ -11,8 +11,8 @@
 // ======================================================================
 
 
-#ifndef DEFRAMING_PROTOCOLINTERFACE_HPP
-#define DEFRAMING_PROTOCOLINTERFACE_HPP
+#ifndef SVC_DEFRAMING_PROTOCOL_INTERFACE_HPP
+#define SVC_DEFRAMING_PROTOCOL_INTERFACE_HPP
 
 #include <Fw/Buffer/Buffer.hpp>
 #include <Fw/Time/Time.hpp>
```
Comment (kevin-f-ortega) [inline Svc/FramingProtocol/DeframingProtocolInterface.hpp:18]: Should this class be inside the Svc namespace?

File: Svc/FramingProtocol/FramingProtocolInterface.hpp [hunk 7]
```diff
@@ -4,18 +4,18 @@
...
-
-#ifndef FRAMING_PROTOCOLINTERFACE_HPP
-#define FRAMING_PROTOCOLINTERFACE_HPP
+#ifndef SVC_FRAMING_PROTOCOL_INTERFACE_HPP
+#define SVC_FRAMING_PROTOCOL_INTERFACE_HPP
 
 #include <Fw/Buffer/Buffer.hpp>
 #include <Fw/Time/Time.hpp>
+
 /**
```
Comment (kevin-f-ortega) [inline Svc/FramingProtocol/FramingProtocolInterface.hpp:19]: This interface should be in the Svc namespace.

**Review by kevin-f-ortega — APPROVED** [review]
LGTM. Good work!

**Review by LeStarch — APPROVED** [review]
Great!

**Pull request merged**